agent/structs/intention.go

```diff
@@ -325,7 +325,7 @@ func (ixn *Intention) CanRead(authz acl.Authorizer) bool {
 }
 
 func (ixn *Intention) CanWrite(authz acl.Authorizer) bool {
-	if authz == nil {
+	if authz == nil || authz == acl.ManageAll() {
```

Comment:
Can the `authz` somehow still be nil here?

Also curious as to what checking against a specific authz singleton provides for us here. IIUC without this check it should fall through and call `authz.IntentionWrite` on line 338 which would be allowed with the `ManageAll` authorizer anyways.

Reply:
I think that's right. I'm going to do a second pass after this to remove more of the nil checks that should no longer be necessary.